### ISSUE TITLE
[DFC Orders] Improve OIDC Error Handling

### ISSUE THREAD

**Issue by RaggedStaff** [openfoodfoundation/openfoodnetwork#12987]
#### :information_source: Funded Feature. Please track ALL ASSOCIATED WORK under the associated tracking code `#11678 DFC Orders`

Need to clean up OIDC processes to provide more meaningful error messging - currently just returning 500 errors. 

Agreed we will never support linking multiple OFN accounts to the same OIDC account (needs error handling/meaningful message)

I (@RaggedStaff ) repeatedly get an error when attempting to link to my Keycloak realm admin account, other accounts work fine. Could this be due to duplication: I have a regular (non-producer/non-admin) OFN account with the same email, but not sure how that would impact ? Can we pass more detail back from the Keycloak response to better understand this errror ?

**Comment by mkllnk** [comment]
> error when attempting to link to my Keycloak realm admin account, other accounts work fine. Could this be due to duplication: I have a regular (non-producer/non-admin) OFN account with the same email, but not sure how that would impact ?

The OFN email address shouldn't matter. It's only a problem if you had multiple OIDC accounts with the same email address.

**Comment by mkllnk** [comment]
> I understand it like this:
> 
>     I have an account on staging FR
>     I have another account on staging UK
> 
> I can't link both accounts to the same OIDC account. Is that correct?


No, if you have two accounts on the same OFN server then you can't link them to the same OIDC account. Otherwise the API doesn't know which of the two user accounts you want to access.

But in your example, you have two different servers and each of them has only one account linked to the same OIDC account. That works and we could use Lescommuns for single-sign-on.

**Comment by mkllnk** [comment]
> If I change the email on OFN can I link all these accounts to the same OIDC account or is the problem elsewhere?

The OFN email address shouldn't matter at all for OIDC here.

**Comment by mkllnk** [comment]
Estimate: 1 day.

**Comment by dacook** [comment]
@mkllnk can you confirm this is the related error? 
https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/66e89c5fdf0c335670df0dd3?filters[event.since]=all
It's the one I experienced [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/13049#issuecomment-2574475265).
You had marked it as fixed, but I guess we still need to address it.

**Comment by mkllnk** [comment]
Yes, that's the error.

**Comment by RaggedStaff** [comment]
@dacook Hopefully not too late... I've replicated the error I ALWAYS get when I try to link my Keycloak admin account to any OFN account on uk-staging (just now).

This is the [error returned](https://staging.openfoodnetwork.org.uk/user/spree_user/auth/openid_connect/callback?state=0f84c68325d421738d22dc161d665ee8&session_state=bca514eb-9f7c-4d2b-a6da-44176db31902&iss=https%3A%2F%2Flogin.lescommuns.org%2Fauth%2Frealms%2Fdata-food-consortium&code=5684ab91-ee62-443f-82ff-442b1dbba3e6.bca514eb-9f7c-4d2b-a6da-44176db31902.53fbf050-85c7-4731-aa14-1bf69c2e6d11).

DM on slack me if you need more details.

**Comment by dacook** [comment]
Thanks @RaggedStaff, but I'm not able to see anything from the link, as it relates to your session. However I can see that an error was logged 2 mins prior to  your message: https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/66ec0f72df0c335670c89854. I'll have a look.
